### PR TITLE
build: parameterise package vsn

### PR DESCRIPTION
**Pull request by zmstone**
We'll likely have to release CE and EE with different git tag schemes (on the same branch).
The least we want is to create another branch for EE release (which would mean there is essentially no difference comparing to the old time when we had two repos), because that's when the code base starts to diverge and becomes hard to trace the changes.
e.g.
`v5.0.0` for CE
`e5.0.0` for EE

### CODE REVIEW

File: .github/workflows/build_packages.yaml [hunk 1]
```diff
@@ -35,10 +36,14 @@ jobs:
...
-          vsn="$(./pkg-vsn.sh)"
-          pre_vsn="$(echo $vsn | grep -oE '^[0-9]+.[0-9]')"
-          old_vsns="$(git tag -l "v$pre_vsn.[0-9]" | xargs echo -n | sed "s/v$vsn//")"
-          echo "::set-output name=old_vsns::$old_vsns"
+          ce_vsn="$(./pkg-vsn.sh community)"
+          ee_vsn="$(./pkg-vsn.sh enterprise)"
+          ce_pre_vsn="$(echo $ce_vsn | grep -oE '^[0-9]+.[0-9]')"
+          ee_pre_vsn="$(echo $ee_vsn | grep -oE '^[0-9]+.[0-9]')"
+          ce_old_vsns="$(git tag -l "v$ce_pre_vsn.[0-9]" | xargs echo -n | sed "s/v$ce_vsn//")"
+          ee_old_vsns="$(git tag -l "e$ee_pre_vsn.[0-9]" | xargs echo -n | sed "s/v$ee_vsn//")"
```
Comment (thalesmg) [inline .github/workflows/build_packages.yaml:44]:
```suggestion
          ee_old_vsns="$(git tag -l "e$ee_pre_vsn.[0-9]" | xargs echo -n | sed "s/e$ee_vsn//")"
```
?

File: .github/workflows/build_packages.yaml [hunk 2]
```diff
@@ -35,10 +36,14 @@ jobs:
...
         working-directory: source
         run: |
-          vsn="$(./pkg-vsn.sh)"
-          pre_vsn="$(echo $vsn | grep -oE '^[0-9]+.[0-9]')"
-          old_vsns="$(git tag -l "v$pre_vsn.[0-9]" | xargs echo -n | sed "s/v$vsn//")"
-          echo "::set-output name=old_vsns::$old_vsns"
+          ce_vsn="$(./pkg-vsn.sh community)"
+          ee_vsn="$(./pkg-vsn.sh enterprise)"
+          ce_pre_vsn="$(echo $ce_vsn | grep -oE '^[0-9]+.[0-9]')"
+          ee_pre_vsn="$(echo $ee_vsn | grep -oE '^[0-9]+.[0-9]')"
```
Comment (thalesmg) [inline .github/workflows/build_packages.yaml:42]: Q: couldn't the minor version number be more than one digit eventually (like in `5.10.0`)?

Comment (zmstone) [reply .github/workflows/build_packages.yaml]: thanks. fixed.
and you just helped to discovered a bug in 4.3
https://github.com/emqx/emqx/pull/6857

**Comment by zmstone** [comment]
the failed build is common test in otp23, succeeded in otp24

File: .github/workflows/build_packages.yaml [hunk 3]
```diff
@@ -35,10 +36,14 @@ jobs:
...
         run: |
-          vsn="$(./pkg-vsn.sh)"
-          pre_vsn="$(echo $vsn | grep -oE '^[0-9]+.[0-9]')"
-          old_vsns="$(git tag -l "v$pre_vsn.[0-9]" | xargs echo -n | sed "s/v$vsn//")"
-          echo "::set-output name=old_vsns::$old_vsns"
+          ce_vsn="$(./pkg-vsn.sh community)"
+          ee_vsn="$(./pkg-vsn.sh enterprise)"
+          ce_base_vsn_prefix="$(echo $ce_vsn | grep -oE '^[0-9]+\.[0-9]+')"
+          ee_base_vsn_prefix="$(echo $ee_vsn | grep -oE '^[0-9]+\.[0-9]+')"
+          ce_old_vsns="$(git tag -l | grep -E "v${ce_base_vsn_prefix}\.[0-9]+$" | grep -v "v${ee_vsn}" | xargs)"
```
Comment (thalesmg) [inline .github/workflows/build_packages.yaml:43]:
```suggestion
          ce_old_vsns="$(git tag -l | grep -E "v${ce_base_vsn_prefix}\.[0-9]+$" | grep -v "v${ce_vsn}" | xargs)"
```
?

**Review by k32 — APPROVED** [review]

**Review by qzhuyan — APPROVED** [review]

**Pull request merged**